### PR TITLE
Merge test harness fixes to the release candidate branch

### PR DESCRIPTION
**Pull request by str4d**
Brings in #908.

### CODE REVIEW

**Review by nuttycom — APPROVED** [review]
utACK; I verified that this merges 101c9c0ea491853b67dda9fdfbf8ae35948df202 which is already in `main` so no need to wait for CI.

**Comment by str4d** [comment]
This is the same commit that was approved in #908, and I verified locally that tests pass after merging. The rustfmt issue is being fixed in #985.

**Review by daira — COMMENTED** [review]
utACK, looks identical to the commits on `main`.

**Pull request merged**
